lib/DataRepeater.js

```diff
@@ -471,6 +473,14 @@ var DataRepeater = module.exports = kind(
 				p.removeListener(e, this._handlers[e]);
 			}
 		}
+		if(c && this.selectionProperty) {
+			var filter = isArray ?
+				function(record) { return record[this.selectionProperty]; } :
+				function(record) { return record.get(this.selectionProperty); };
```

Comment:
functions should be bound to the scope to identify the scope's properties inside I guess.